DigitalOffice/Tests/FileServiceTests/Commands/AddNewFileCommandTests.cs

```diff
@@ -0,0 +1,109 @@
+﻿using FileService.Commands;
+using FileService.Commands.Interfaces;
+using FileService.Database.Entities;
+using FileService.Mappers.Interfaces;
+using FileService.Models;
+using FileService.Repositories.Interfaces;
+using FluentValidation;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Security.Policy;
+
+namespace FileServiceUnitTests.Commands
+{
+    public class AddNewFileCommandTests
+    {
+        private DbFile newFile;
+        private FileRequest fileRequest;
+        private IAddNewFileCommand command;
+        private Mock<IFileRepository> repositoryMock;
+        private Mock<IValidator<FileRequest>> validatorMock;
+        private Mock<IMapper<FileRequest, DbFile>> mapperMock;
+
+        [SetUp]
+        public void Initialization()
+        {
+            repositoryMock = new Mock<IFileRepository>();
+            validatorMock = new Mock<IValidator<FileRequest>>();
+            mapperMock = new Mock<IMapper<FileRequest, DbFile>>();
+
+            newFile = new DbFile
+            {
+                Id = Guid.NewGuid(),
+                Content = Convert.FromBase64String("RGlnaXRhbCBPZmA5Y2U="),
+                Extension = ".txt",
+                IsActive = true,
+                Name = "DigitalOfficeTestFile",
+                AddedOn = DateTime.UtcNow
+            };
+
+            fileRequest = new FileRequest
+            {
+                Content = "RGlnaXRhbCBPZmA5Y2U=",
+                Extension = ".txt",
+                Name = "DigitalOfficeTestFile"
+            };
+
+            mapperMock
+                .Setup(f => f.Map(It.IsAny<FileRequest>()))
+                .Returns(newFile);
+            command = new AddNewFileCommand(repositoryMock.Object, validatorMock.Object, mapperMock.Object);
+        }
+
+        [Test]
+        public void SuccessfulAddNewFileTest()
+        {
+            var fileId = Guid.NewGuid();
+
+            validatorMock
+                 .Setup(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                 .Returns(true);
+
+            repositoryMock
+                .Setup(x => x.AddNewFile(It.IsAny<DbFile>()))
+                .Returns(fileId);
+
+            Assert.AreEqual(fileId, command.Execute(fileRequest));
```

Comment:
It would be a good idea to also check that when the command.Execute is called, the corresponding methods of the mocks are called.

Reply:
Could please advise me how to do this?
I cannot find it in your Pull Request (GetUser)

Reply:
Well, I have such check in my pull request. Example with mapper:
`mapperMock.Verify(mapper=> mapper.Map(fileReqeust), Times.Once);`
Similarly with other mocks.

Reply:
Huh. Weird, how'd I miss this. Thanks for info, will fix

Reply:
Fixed